src/Common/HashTable/BloomFilt.h

```diff
@@ -0,0 +1,118 @@
+#pragma once
+
+#include <functional>
+#include <cmath>
+//#include <MurmurHash3.h>
+//#include <Functions/FunctionsHashing.h>
+#include <city.h>
+#include <random>
+#include <Common/HashTable/HashTableAllocator.h>
+
+template <typename T, typename Hash, typename Allocator = HashTableAllocator>
+class BloomFilt : protected Hash,
+                protected Allocator
+{
+public:
+    BloomFilt(size_t max_item, double  = 0.001);
+
+    size_t hash(const T & x) const { return Hash::operator()(x); }
+
+    //void init(size_t max_item, double p = 0.01);
+
+    void insert(const T & elem);
+
+    bool lookup(const T & elem) const;
+
+    double bitsPerItem() const;
+
+    size_t getBufferSizeInBytes() const;
+
+    ~BloomFilt() {
+        if (array)
+        {
+            Allocator::free(array, size_of_filter * sizeof(bool));
+            array = nullptr;
+        }
+    }
+
+private:
+    static constexpr UInt64 SEED_GEN_A = 845897321;
+    static constexpr UInt64 SEED_GEN_B = 217728422;
+
+    size_t seed;
+    std::array<uint64_t, 2> _hash(size_t elem) const;
+    uint64_t nthHash(size_t n,
+                        uint64_t hashA,
+                        uint64_t hashB) const;
+
+    const double size_param = log(2.0);
+    const size_t min_size = 10;
+    size_t size_of_filter, number_of_hash, number_of_elem;
+    bool * array = nullptr;
```

Comment:
Consider using `PaddedPODArray<bool>` as array of boolean instead of manual memory management.